extension/asapclientauthextension/extension_test.go

```diff
@@ -0,0 +1,110 @@
+package asapclientauthextension
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"bitbucket.org/atlassian/go-asap"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// mockRoundTripper copies the request headers to the response.
+// So, the caller can make assertions using the returned response.
+type mockRoundTripper struct{}
```

Comment:
I suggest you have a look at https://pkg.go.dev/github.com/stretchr/testify/mock since it be more explicit as you adopt it in your test cases.

Reply:
I copied the mockroundtripper from the bearer auth extension, so may as well keep it consistent across the project.

Reply:
Make sense to me :)